includes/Classifai/Providers/Azure/Language.php

```diff
@@ -0,0 +1,391 @@
+	 * @param array  $args Optional arguments to pass to the route.
+	 * @return string|WP_Error
+	 */
+	public function rest_endpoint_callback( $post_id = 0, string $route_to_call = '', array $args = [] ) {
+		if ( ! $post_id || ! get_post( $post_id ) ) {
+			return new WP_Error( 'post_id_required', esc_html__( 'A valid post ID is required to generate an excerpt.', 'text-domain' ) );
+		}
+
+		$route_to_call = strtolower( $route_to_call );
+		$return        = '';
+
+		switch ( $route_to_call ) {
+			case 'excerpt':
+				$return = $this->generate_excerpt( $post_id, $args );
+				break;
+		}
+
+		return $return;
+	}
+
+	/**
+	 * Generate an excerpt for a given post.
+	 *
+	 * This service requires two API calls - one to request the summary and another to retrieve the summary.
+	 *
+	 * @param int   $post_id The Post ID we're processing.
+	 * @param array $args Optional arguments to pass to the route.
+	 * @return string
+	 */
+	public function generate_excerpt( $post_id, $args ) {
+		$feature      = new ExcerptGeneration();
+		$settings     = $feature->get_settings();
+		$api_key      = $settings[ static::ID ]['api_key'];
+		$endpoint_url = $settings[ static::ID ]['endpoint_url'];
+		$post_content = get_post_field( 'post_content', $post_id );
+
+		// Request the summary form the API.
+		$summary_request_url = $this->request_summary( $endpoint_url, $api_key, $post_content, $post_id );
+		if ( is_wp_error( $summary_request_url ) ) {
+			return $summary_request_url;
+		}
+
+		// Retrieve the Summary after the job is complete.
+		$summary = $this->retrieve_summary( $summary_request_url );
+
+		return $summary;
+	}
+
+	/**
+	 * Request the summary from the API.
+	 *
+	 * @param string $endpoint_url The endpoint URL.
+	 * @param string $api_key The API key.
+	 * @param string $post_content The post content.
+	 * @param int    $post_id The post ID.
+	 *
+	 * @return mixed The summary job URL or WP_Error.
+	 */
+	public function request_summary( $endpoint_url, $api_key, $post_content, $post_id ) {
+		$endpoint_url = add_query_arg(
+			'api-version',
+			static::API_VERSION,
+			$endpoint_url . static::ANALYZE_TEXT_ENDPOINT
+		);
+
+		$body = [
+			'analysisInput' => [
+				'documents' => [
+					[
+						'id'       => '1',
+						'language' => 'en',
+						'text'     => $post_content,
+					],
+				],
+			],
+			'tasks'         => [
+				[
+					'kind'       => 'AbstractiveSummarization',
+					'taskName'   => 'Classifai Summarization ' . $post_id,
+					'parameters' => [
+						/**
+						 * Filter the summary length.
+						 * Possible values are 'oneSentence', 'short', 'medium', 'long'.
+						 * Default is 'oneSentence'.
+						 *
+						 * @since 3.2.0
+						 * @hook classifai_azure_language_summary_length
+						 * @param string $summary_length The summary length.
+						 * @return string
+						 */
+						'summaryLength' => apply_filters( 'classifai_azure_language_summary_length', 'oneSentence' ),
```

Comment:
We'll need to format the doc block

```php
/**
 * Filter the summary length.
 * Possible values are 'oneSentence', 'short', 'medium', 'long'.
 * Default is 'oneSentence'.
 *
 * @since 3.2.0
 * @hook classifai_azure_language_summary_length
 * @param {string} $summary_length The summary length.
 * @return {string} The summary length
 */
```